File: Programming Language/Python3/is-graph-bipartite.py
class Solution:
    def isBipartite(self, graph: List[List[int]]) -> bool:
        n = int(len(graph))
        color = [-1]*n
        for i in range(0, n):
            if color[i] == -1:
                self.ok = True
                self.dfs(i, 0, color, graph)
                if self.ok == False:
                    return False;
        return True;
    def dfs(self, u, c, color, graph):
        color[u] = c
        for v in graph[u]:
            if color[v] == -1:
                self.dfs(v, c^1, color, graph)
            else:
                if color[u] == color[v]:
                    self.ok = False

File: Programming Language/Java/is-graph-bipartite.java
public class Solution {
    boolean ok = true;
    public boolean isBipartite(int[][] graph) {
        int n = graph.length;
        int[] color = new int[n];
        for(int i = 0; i < n; i++){
            color[i] = -1;
        }
        for(int i = 0; i < n; i++){
            if(color[i] == -1){
                ok = true;
                dfs(i, 0, color, graph);
                if(ok == false) return false;
            }
        }
        return true;
    }
    public void dfs(int u, int c, int[] color, int[][] graph){
        color[u] = c;
        for(int v: graph[u]){
            if(color[v] == -1){
                dfs(v, c^1, color, graph);
            }else{
                if(color[u] == color[v]){
                    ok = false;
                }
            }
        }
    }
}
